src/Elastic.Transport/Products/Elasticsearch/ElasticsearchProductRegistration.cs

```diff
@@ -31,10 +33,11 @@ public class ElasticsearchProductRegistration : ProductRegistration
 	/// 
 	/// </summary>
 	/// <param name="markerType"></param>
-	public ElasticsearchProductRegistration(Type markerType) : this()
+	/// <param name="serviceIdentifier"></param>
+	public ElasticsearchProductRegistration(Type markerType, string serviceIdentifier = ElasticsearchServiceIdentifier) : this()
```

Comment:
A personal preference here for public ctors/methods would be to add a ctor overload rather than an optional parameter. I believe this aligns with the .NET API guidance for public methods as it's more flexible to avoid breaking changes.

I'm also wondering if an alternative design might be considered here if we expect all future products to include a meta header. If that's likely, we could add an abstract string property to the `ProductRegistration`, which returns a service identifier. If null, perhaps we don't configure a `MetaHeaderProvider` at all. Then `ElasticsearchProductRegistration` could be implemented with a virtual property, and any derived product types could override if required.

Reply:
> I'm also wondering if an alternative design might be considered here if we expect all future products to include a meta header. 

Not quite sure about this, but I like the property solution anyways 🙂 Will change the code accordingly.